plugin/adsk/scripts/USDMenuProc.mel

```diff
@@ -40,16 +62,16 @@ global proc USDMenuProc(string $parent, string $obj)
         setParent -menu $parent;
 
         setParent -menu ..;
-        if (canEditAsMaya($obj)) {
+        if (canEditAsMaya()) {
             // Temporary - hide this context menu items behind an
             //             env var until it is completed.
             string $s = `getenv "MAYAUSD_ENABLE_EDIT_AS_MAYA_DATA"`;
             string $ls = tolower($s);
             if (`match $ls "true|yes|on|1"` !="")
             {
-                menuItem -label "Edit As Maya Data" -image "edit_as_Maya.png" -command ("mayaUsdEditAsMaya \"" + $obj + "\"");
+                menuItem -label "Edit As Maya Data" -image "edit_as_Maya.png" -command ("mayaUsdMenu_editAsMaya");
             }
         }
-        menuItem -label "Duplicate As Maya Data" -command ("mayaUsdDuplicate \"" + $obj + "\"" + "\"|world\"");
+        menuItem -label "Duplicate As Maya Data" -command ("mayaUsdMenu_duplicate");
```

Comment:
As a general rule the way these things normally work is if the input object is selected, then you operate on the entire selection, otherwise you operate on only the input object. Does that work in this case?

Reply:
IIUC, if $obj is in the selection, use the code I wrote, else use only $obj?

My mel knowledge is low, how would I check if $obj is selected?

Reply:
Found:
`proc int isObjectSelected(string $obj)
{
    if (size($obj)) {
        string $sel[] = `ls -sl`;
        return stringArrayContains($obj, $sel);
    }
    return 0;
}
`

Reply:
I tried it, but the object is not part of the selection even though it is part of the selection. 😃 What I mean is that when you select the parent of geom in the outliner, the whole object hierarchy is selected in the viewport, but the selection as returned by `ls -sl -ufe` only contains the top node.

So, I cannot do as you suggested. The goal is to edit the UFE selection, even if the user right-click on a sub-part of the object in the viewport.

Reply:
What you are getting as the input $obj is the Ufe path string for the object. So you need the `-ufe` flag to `ls`. Something like this:
```
proc int isObjectSelected(string $obj) { if (size($obj)) { string $sel[] = `ls -sl -ufe`; return stringArrayContains($obj, $sel); } return 0; }
```

But it looks like this proc only gets called when there is a non-Maya item selected. In fact you don't even need to RMB over the non-Maya item (in the viewport) for this menu to be called. Simply RMB over empty space. So I guess the input object is really irrelevant. It would just be the first non-Maya item selected and not necessarily the one you RMB over.

So you can ignore my first comment.